app/scripts/controllers/courtSearchInfo.ctrl.js

```diff
@@ -17,21 +18,20 @@ angular.module('yourStlCourts').controller('courtSearchInfoCtrl', function ($sta
     shadowSize: [41, 41]
   };
 
-
-
-
-
-
-  if(!courtInfo) {
+  if(!courtId) {
     $state.go('home');
   } else {
-    ctrl.courtInfo = courtInfo;
-    ctrl.courtDirectionLink = getCourtDirectionLink(courtInfo);
-    ctrl.courtOnMap.lat = courtInfo.latitude;
-    ctrl.courtOnMap.lng = courtInfo.longitude;
-    ctrl.courtMarkers.m1 = {lat:courtInfo.latitude,lng:courtInfo.longitude, message: "Test",icon: courtDefaultIcon};
-    //ctrl.courtMarker.lat = courtInfo.latitude;
-    //ctrl.courtMarker.lng = courtInfo.longitude;
+    Courts.findById(courtId).then(function(court){
+      if (court.id === undefined){
+        $state.go('home');
+      }else {
```

Comment:
court.id should never be undefined. If the request is for a Court with an invalid id, the promise will reject which is the second function callback parameter of the ".then" function. This is typically where we show a toastr error message to the user.